src/schedule/schedule.service.ts

```diff
@@ -1,26 +1,62 @@
-import { Injectable } from '@nestjs/common';
 import { CreateScheduleDto } from './dto/create-schedule.dto';
 import { UpdateScheduleDto } from './dto/update-schedule.dto';
+import { InjectModel } from '@nestjs/mongoose';
+import { ScheduleInterface } from './interfaces/schedule.interface';
+import { Model } from 'mongoose';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 
 @Injectable()
 export class ScheduleService {
-  create(createScheduleDto: CreateScheduleDto) {
-    return 'This action adds a new schedule';
+  constructor(@InjectModel('Schedule') private readonly scheduleModel: Model<ScheduleInterface>) {}
+
+  async create(createScheduleDto: CreateScheduleDto) {
+    const schedule = await new this.scheduleModel(createScheduleDto);
+    return schedule.save();
+  }
+
+  async findAll() {
+    const schedules = await this.scheduleModel.find().exec();
+    if (!schedules || !schedules[0]) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return schedules;
   }
 
-  findAll() {
-    return `This action returns all schedule`;
+  async findOne(id: string) {
+    const schedule = await this.scheduleModel.findOne({ _id: id }).exec();
+    if (!schedule) {
+      throw new HttpException('Not Found', HttpStatus.NOT_FOUND);
+    }
+    return schedule;
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} schedule`;
+  async findScheduleByLesson(id: string) {
+    const schedules = await this.scheduleModel.find().exec();
+
+    schedules.forEach((schedule) => {
+      schedule.lessons.forEach((lessonId) => {
```

Comment:
tutaj trzeba będzie to sprawdzać po id z bazy danych, które dodaje mongoose, czyli np. `forEach((lesson) => if (lesson._id ==id)` , to `_id` jest właśnie tym które tworzy mongoose